### PR TITLE
feat(github-action-plan-storage): added

### PR DESCRIPTION
**Pull request by dudymas**
## what
* terraform plan storage for github actions

## why
* supports [github-action-terraform-plan-storage](https://github.com/cloudposse/github-action-terraform-plan-storage)

## references

image

### CODE REVIEW

File: modules/github-action-plan-storage/iam.tf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  }
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:PutObject",
+      "s3:GetObject",
+      "s3:DeleteObject",
+    ]
+    resources = [
+      "${module.tfstate_backend.tfstate_backend_s3_bucket_arn}/*",
```
Comment (mcalhoun) [inline modules/github-action-plan-storage/iam.tf:37]: I think this also needs `${module.tfstate_backend.tfstate_backend_s3_bucket_arn}` (without the splat) in order for the ListBucket to work properly.

**Review by mcalhoun — CHANGES_REQUESTED** [review]
Looks like a great start, thanks! I left a few comments.

File: modules/github-action-plan-storage/iam.tf [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    ]
+  }
+  statement {
+    effect = "Allow"
+    actions = [
+      "dynamodb:DescribeTable",
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:DeleteItem"
```
[the remark this answers is not in the document]
Comment (dudymas) [reply modules/github-action-plan-storage/iam.tf:46]: gotcha. fixing

File: modules/github-action-plan-storage/main.tf
[the remark this answers is not in the document]
Comment (dudymas) [reply modules/github-action-plan-storage/main.tf:1]: adding that neow

File: modules/github-action-plan-storage/main.tf [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  enabled = var.enabled
+  context = module.this.context
+  region  = var.region
+
+  enable_server_side_encryption = var.enable_server_side_encryption
+  enable_point_in_time_recovery = var.enable_point_in_time_recovery
+  force_destroy                 = var.force_destroy
+  prevent_unencrypted_uploads   = var.prevent_unencrypted_uploads
+  access_roles                  = var.access_roles
+}
```
Comment (osterman) [inline modules/github-action-plan-storage/main.tf:52]: It's considered bad practice to source root modules as child modules. Root modules have a state backend.

**Comment by dudymas** [comment]
Closing as stale

**Pull request closed without merging**